opm/autodiff/SimulatorFullyImplicitBlackoilOutput.hpp

```diff
@@ -452,15 +457,18 @@ namespace Opm
 
         const Wells* wells = wellsmanager.c_wells();
         wellstate.resize(wells, simulatorstate, phaseUsage ); //Resize for restart step
-        auto state = eclIO_->loadRestart(solution_keys);
+        auto restart_values = eclIO_->loadRestart(solution_keys, extra_keys);
 
-        solutionToSim( state.first, phaseUsage, simulatorstate );
-        wellsToState( state.second, phaseUsage, wellstate );
+        solutionToSim( restart_values.solution, phaseUsage, simulatorstate );
+        wellsToState( restart_values.wells, phaseUsage, wellstate );
 
-        if (state.first.has( "OPMEXTRA" ) ) {
-            std::vector<double> opmextra = state.first.data( "OPMEXTRA" );
+        const auto opmextra_iter = restart_values.extra.find("OPMEXTRA");
+        if (opmextra_iter != restart_values.extra.end()) {
+            std::vector<double> opmextra = opmextra_iter->second;
             assert(opmextra.size() == 1);
             extra.suggested_step = opmextra[0];
+        } else {
+            OPM_THROW(std::runtime_error, "Cannot restart, restart data is missing OPMEXTRA field.");
```

Comment:
Fair enough to have this here - but in the current implementation the output layer has already thrown `std::runtime_error()` if the `OPMEXTRA` field could not be loaded.